text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,1678 @@
+            4. Supports post-processing automation
+                * After scaffolding, Yeoman can automatically run `npm install`, initialize Git, and add license headers
+        * Cons:
+            1. Increased complexity and learning curve
+                * CDK users would need to install and learn Yeoman
+                * Template authors must write JS-based generators instead of simple file templates
+            2. External dependency
+                * Relies on the Yeoman runtime and ecosystem so CDK loses control over some UX
+                * Added maintenance risk from external changes
+            3. Not natively integrated into CDK CLI
+                * `cdk init` logic needs to be updated to recognize and delegate to Yeoman generators
+                * Makes CDK’s bootstrapping process more complex
+    4. Projen
+        * Pros: 
+            1. Controlled and maintained by AWS
+                * Easier to add CDK specific behaviors
+            2. Automates common setup tasks (like linting, testing, and publishing configs)
+                * Template authors don’t have to maintain those manually across many templates.
+            3. One Projen-based template can generate consistent setups across different language targets.
+        * Cons: 
+            1. Steep learning curve
+                * Template authors must understand how Projen works, including its configuration language
+            2. Templates are code, not files
+                * Instead of writing/copying static template files, template authors must write Projen config code that generates those files.
+            3. Generated files are not meant to be edited
+                * Generated files are overwrited on every synth, so template users should only modify the config and not the output file
+            4. Harder to preview template
+                * Since the actual files aren’t visible until you run `projen synth`, don’t initially know what the final output will look like.
+                * Can slow down the template authoring and review process.
+    5. Template authors develop and maintain their own placeholder and template configuration
+        * Pros:
+            1. Simpler from a user point of view because valid CLI arguments are more concrete
+            2. cdk init does not have to support template configuration
+                * If template authors want to have variations on templates, they can do the branching generation-side, which allows them to use more advanced tools
+        * Cons:
+            1. No default placeholder or template configuration style for template authors
+
+
+
+### What are the drawbacks of this solution?
+
+1. Template validation burden is on template author, can't ensure template quality
+2. Current proposal only supports static templates and not dynamic templates generated from user input
```

Comment:
We determined that this is currently not something users want/need - so why is it a drawback? also - does the solution prevent supporting dynamic templates in the future?